packages/react-paypal-js/src/v6/hooks/usePayPalGuestPaymentSession.ts

```diff
@@ -0,0 +1,140 @@
+          createOrder: () => PayPalGuestOneTimePaymentSessionPromise;
+          orderId?: never;
+      })
+    | (PayPalGuestOneTimePaymentSessionOptions & {
+          createOrder?: never;
+          orderId: string;
+      })
+) &
+    PayPalGuestPresentationModeHookOptions;
+
+export function usePayPalGuestPaymentSession({
+    fullPageOverlay,
+    createOrder,
+    orderId,
+    targetElement,
+    onShippingAddressChange,
+    onShippingOptionsChange,
+    ...callbacks
+}: UsePayPalGuestPaymentSessionProps): PayPalGuestPaymentSessionReturn {
+    const { sdkInstance } = usePayPal();
+    const isMountedRef = useIsMountedRef();
+    const sessionRef = useRef<PayPalGuestOneTimePaymentSession | null>(null);
+    const buttonRef = useRef<HTMLElement>(null);
+    const proxyCallbacks = useProxyProps(callbacks);
+    const [error, setError] = useError();
+    const isSessionActiveRef = useRef(false);
+
+    const handleDestroy = useCallback(() => {
+        sessionRef.current?.destroy();
+        sessionRef.current = null;
+        isSessionActiveRef.current = false;
+    }, []);
+
+    useEffect(() => {
+        if (!sdkInstance) {
+            setError(new Error("no sdk instance available"));
+        }
+    }, [sdkInstance, setError]);
+
+    useEffect(() => {
+        if (!sdkInstance) {
+            return;
+        }
+
+        const newSession = sdkInstance.createPayPalGuestOneTimePaymentSession({
+            orderId,
+            ...proxyCallbacks,
+            ...(onShippingAddressChange && { onShippingAddressChange }),
+            ...(onShippingOptionsChange && { onShippingOptionsChange }),
+        });
+        sessionRef.current = newSession;
+
+        return () => {
+            newSession.destroy();
+            isSessionActiveRef.current = false;
+        };
+    }, [
+        sdkInstance,
+        orderId,
+        proxyCallbacks,
+        onShippingAddressChange,
+        onShippingOptionsChange,
+        isMountedRef,
+    ]);
+
+    const handleCancel = useCallback(() => {
+        isSessionActiveRef.current = false;
+    }, []);
+
+    const handleClick = useCallback(async () => {
+        if (!isMountedRef.current) {
+            return;
+        }
+        if (!sessionRef.current) {
+            setError(new Error("PayPal Guest Checkout session not available"));
+            return;
+        }
+
+        if (isSessionActiveRef.current) {
+            return;
+        }
+
+        isSessionActiveRef.current = true;
+
+        try {
+            const target = targetElement || buttonRef.current;
+            const startOptions: PayPalGuestPresentationModeOptions = {
+                presentationMode: "auto",
+                ...(fullPageOverlay !== undefined && { fullPageOverlay }),
+                ...(target ? { targetElement: target as EventTarget } : {}),
+            };
+            const checkoutOptionsPromise = createOrder
+                ? createOrder()
+                : undefined;
```

Comment:
Using optional chaining, I believe this could be shortened to:

```suggestion
            const checkoutOptionsPromise = createOrder?.();
```

This could be moved down to line 123 as well, same as the PayLater session is doing:

https://github.com/paypal/paypal-js/blob/main/packages/react-paypal-js/src/v6/hooks/usePayLaterOneTimePaymentSession.ts#L120